src/appleseed/foundation/math/sampling/mappings.h

```diff
@@ -558,7 +558,7 @@ inline T exponential_distribution_on_segment_pdf(
     const T l,
     const T r)
 {
-    assert (x >= l && x < r);
+    assert (x >= l && x <= r);
```

Comment:
Mathematically, the segment [l, r) fits better. But programmatically...
Due to imprecise calculations, the actual values are generated in (l - eps, r + eps), and I use `clamp` in `sample` to put values in the required range. Naturally, `clamp` gives me values in [l, r].
Probably, that should be reflected in the description, yes.